diff --git a/drivers/net/ixgbe/base/ixgbe_82599.c b/drivers/net/ixgbe/base/ixgbe_82599.c
index 7166d04..7a522f2 100644
--- a/drivers/net/ixgbe/base/ixgbe_82599.c
+++ b/drivers/net/ixgbe/base/ixgbe_82599.c
@@ -2456,6 +2456,9 @@ STATIC s32 ixgbe_read_eeprom_buffer_82599(struct ixgbe_hw *hw, u16 offset,
 	return ret_val;
 }
 
+void do_not_optimize_this_value_out_please(int val) __attribute__((noinline));
+
+
 /**
  *  ixgbe_read_eeprom_82599 - Read EEPROM word using
  *  fastest available method
@@ -2474,6 +2477,11 @@ STATIC s32 ixgbe_read_eeprom_82599(struct ixgbe_hw *hw,
 
 	DEBUGFUNC("ixgbe_read_eeprom_82599");
 
+#ifdef VIGOR_EXECUTABLE
+        //orig_printf("eeprom->type = %d while ixgbe_eeprom_spi is %d\n", eeprom->type, ixgbe_eeprom_spi);
+        do_not_optimize_this_value_out_please(eeprom->type);
+#endif//VIGOR_EXECUTABLE
+        //orig_printf("eeprom->type = %d while ixgbe_eeprom_spi is %d\n", eeprom->type, ixgbe_eeprom_spi);
 	/*
 	 * If EEPROM is detected and can be addressed using 14 bits,
 	 * use EERD otherwise use bit bang
diff --git a/drivers/net/ixgbe/base/ixgbe_common.c b/drivers/net/ixgbe/base/ixgbe_common.c
index a93caff..e448714 100644
--- a/drivers/net/ixgbe/base/ixgbe_common.c
+++ b/drivers/net/ixgbe/base/ixgbe_common.c
@@ -1263,6 +1263,7 @@ s32 ixgbe_init_eeprom_params_generic(struct ixgbe_hw *hw)
 		 * If not present leave as none
 		 */
 		eec = IXGBE_READ_REG(hw, IXGBE_EEC_BY_MAC(hw));
+                //orig_printf("read eec: %u\n", eec);
 		if (eec & IXGBE_EEC_PRES) {
 			eeprom->type = ixgbe_eeprom_spi;
 
@@ -4102,6 +4103,8 @@ s32 ixgbe_set_vlvf_generic(struct ixgbe_hw *hw, u32 vlan, u32 vind,
 	return IXGBE_SUCCESS;
 }
 
+void do_not_optimize_this_value_out_please(int val) __attribute__((noinline));
+
 /**
  *  ixgbe_clear_vfta_generic - Clear VLAN filter table
  *  @hw: pointer to hardware structure
@@ -4114,6 +4117,10 @@ s32 ixgbe_clear_vfta_generic(struct ixgbe_hw *hw)
 
 	DEBUGFUNC("ixgbe_clear_vfta_generic");
 
+#ifdef VIGOR_EXECUTABLE
+        //orig_printf("vft_size for %p is %d", hw, hw->mac.vft_size);
+        do_not_optimize_this_value_out_please(hw->mac.vft_size);
+#endif//VIGOR_EXECUTABLE
 	for (offset = 0; offset < hw->mac.vft_size; offset++)
 		IXGBE_WRITE_REG(hw, IXGBE_VFTA(offset), 0);
 
diff --git a/drivers/net/ixgbe/base/ixgbe_osdep.h b/drivers/net/ixgbe/base/ixgbe_osdep.h
index bb5dfd2..26b1d79 100644
--- a/drivers/net/ixgbe/base/ixgbe_osdep.h
+++ b/drivers/net/ixgbe/base/ixgbe_osdep.h
@@ -126,7 +126,7 @@ typedef int		bool;
 
 #define IXGBE_PCI_REG(reg) rte_read32(reg)
 
-static inline uint32_t ixgbe_read_addr(volatile void* addr)
+static uint32_t __attribute__((noinline)) ixgbe_read_addr(volatile void* addr)
 {
 	return rte_le_to_cpu_32(IXGBE_PCI_REG(addr));
 }
diff --git a/lib/librte_eal/common/include/generic/rte_io.h b/lib/librte_eal/common/include/generic/rte_io.h
index 0b88c34..3c11bfe 100644
--- a/lib/librte_eal/common/include/generic/rte_io.h
+++ b/lib/librte_eal/common/include/generic/rte_io.h
@@ -328,7 +328,7 @@ rte_read16(const volatile void *addr)
 	return val;
 }
 
-static __rte_always_inline uint32_t
+static inline uint32_t
 rte_read32(const volatile void *addr)
 {
 	uint32_t val;
