diff --git a/lib/librte_eal/common/eal_common_timer.c b/lib/librte_eal/common/eal_common_timer.c
index 568ae2f..b71aa3b 100644
--- a/lib/librte_eal/common/eal_common_timer.c
+++ b/lib/librte_eal/common/eal_common_timer.c
@@ -97,7 +97,7 @@ void rte_delay_us_callback_register(void (*userfunc)(unsigned int))
        rte_delay_us = userfunc;
 }
 
-RTE_INIT(rte_timer_init)
+RTE_INIT_PRIO(rte_timer_init, LOG)
 {
        /* set rte_delay_us_block as a delay function */
        rte_delay_us_callback_register(rte_delay_us_block);
